File: Python3/프로그래머스/2/132265. 롤케이크 자르기/롤케이크 자르기.py
from collections import Counter

def solution(topping):
    answer = 0
    older, younger = {i : 0 for i in set(topping)}, dict(Counter(topping))
    o_cnt, y_cnt = 0, len(younger)
    
    for top in topping:
        older[top] += 1
        if older[top] == 1:
            o_cnt += 1
        
        younger[top] -= 1
        if younger[top] == 0:
            y_cnt -= 1
        
        # print("Older Count:", o_cnt)
        # print("Younger Count:", y_cnt)
        if o_cnt == y_cnt:
            answer += 1
    
    return answer

File: C++/프로그래머스/2/132265. 롤케이크 자르기/롤케이크 자르기.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

int solution(vector<int> topping) {
    int answer = 0;
    unordered_map<int, int> older, younger;
    int o_cnt = 0, y_cnt = 0;
    for(auto& top : topping) {
        if(younger[top]++ == 0) {
            y_cnt++;
        }
        older[top] = 0;
    }
    
    for(auto& top : topping) {
        if(older[top]++ == 0) {
            o_cnt++;
        }
        if(--younger[top] == 0) {
            y_cnt--;
        }
        
        answer += (o_cnt == y_cnt);
    }
    
    return answer;
}